packages/caver-transaction/src/index.d.ts

```diff
@@ -0,0 +1,166 @@
+import ValueTransfer from './transactionTypes/valueTransfer/valueTransfer'
+import FeeDelegatedValueTransferMemo from './transactionTypes/valueTransferMemo/feeDelegatedValueTransferMemo'
+import FeeDelegatedValueTransferMemoWithRatio from './transactionTypes/valueTransferMemo/feeDelegatedValueTransferMemoWithRatio'
+import ValueTransferMemo from './transactionTypes/valueTransferMemo/valueTransferMemo'
+
+export type Transaction =
+    | LegacyTransaction
+    | ValueTransfer
+    | FeeDelegatedValueTransfer
+    | FeeDelegatedValueTransferWithRatio
+    | ValueTransferMemo
+    | FeeDelegatedValueTransferMemo
+    | FeeDelegatedValueTransferMemoWithRatio
+    | AccountUpdate
+    | FeeDelegatedAccountUpdate
+    | FeeDelegatedAccountUpdateWithRatio
+    | SmartContractDeploy
+    | FeeDelegatedSmartContractDeploy
+    | FeeDelegatedSmartContractDeployWithRatio
+    | SmartContractExecution
+    | FeeDelegatedSmartContractExecution
+    | FeeDelegatedSmartContractExecutionWithRatio
+    | Cancel
+    | FeeDelegatedCancel
+    | FeeDelegatedCancelWithRatio
+    | ChainDataAnchoring
+    | FeeDelegatedChainDataAnchoring
+    | FeeDelegatedChainDataAnchoringWithRatio
+
+export default interface ITransaction {
+    decode: typeof TransactionDecoder['decode']
+    getTransactionByHash(transactionHash: string): AbstractTransaction
+    recoverPublicKeys(rawTx: string): string[]
+    recoverFeePayerPublicKeys(rawTx: string): string[]
+    legacyTransaction: typeof LegacyTransaction
+    valueTransfer: typeof ValueTransfer
+    feeDelegatedValueTransfer: typeof FeeDelegatedValueTransfer
+    feeDelegatedValueTransferWithRatio: typeof FeeDelegatedValueTransferWithRatio
+    valueTransferMemo: typeof ValueTransferMemo
+    feeDelegatedValueTransferMemo: typeof FeeDelegatedValueTransferMemo
+    feeDelegatedValueTransferMemoWithRatio: typeof FeeDelegatedValueTransferMemoWithRatio
+    accountUpdate: typeof AccountUpdate
+    feeDelegatedAccountUpdate: typeof FeeDelegatedAccountUpdate
+    feeDelegatedAccountUpdateWithRatio: typeof FeeDelegatedAccountUpdateWithRatio
+    smartContractDeploy: typeof SmartContractDeploy
+    feeDelegatedSmartContractDeploy: typeof FeeDelegatedSmartContractDeploy
+    feeDelegatedSmartContractDeployWithRatio: typeof FeeDelegatedSmartContractDeployWithRatio
+    smartContractExecution: typeof SmartContractExecution
+    feeDelegatedSmartContractExecution: typeof FeeDelegatedSmartContractExecution
+    feeDelegatedSmartContractExecutionWithRatio: typeof FeeDelegatedSmartContractExecutionWithRatio
+    cancel: typeof Cancel
+    feeDelegatedCancel: typeof FeeDelegatedCancel
+    feeDelegatedCancelWithRatio: typeof FeeDelegatedCancelWithRatio
+    chainDataAnchoring: typeof ChainDataAnchoring
+    feeDelegatedChainDataAnchoring: typeof FeeDelegatedChainDataAnchoring
+    feeDelegatedChainDataAnchoringWithRatio: typeof FeeDelegatedChainDataAnchoringWithRatio
+
+    type: {
+        TxTypeLegacyTransaction: 'TxTypeLegacyTransaction'
```

Comment:
This tag string and tag hex number string are also duplicated. ;-;

Reply:
I abstracted to key and values. please check c46afc4